delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/disk/DiskStorageAccess.scala

```diff
@@ -1,28 +1,28 @@
 package ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.disk
 
+import cats.effect.IO
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageRejection.StorageNotAccessible
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.model.StorageValue.DiskStorageValue
 import ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.StorageAccess
 import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
-import monix.bio.{IO, UIO}
 
 import java.nio.file.Files
 
 object DiskStorageAccess extends StorageAccess {
   override type Storage = DiskStorageValue
 
-  override def apply(id: Iri, storage: DiskStorageValue): IO[StorageNotAccessible, Unit] = {
+  override def apply(id: Iri, storage: DiskStorageValue): IO[Unit] = {
 
-    def failWhen(condition: Boolean, err: => String) =
-      if (condition) IO.raiseError(StorageNotAccessible(id, err))
-      else IO.unit
+    def failWhen(condition: Boolean, err: => String) = {
+      IO.raiseWhen(condition)(StorageNotAccessible(id, err))
+    }
 
     for {
-      exists      <- UIO.delay(Files.exists(storage.volume.value))
+      exists      <- IO.delay(Files.exists(storage.volume.value))
```

Comment:
There's a nice functional wrapper lib for file operations  - https://github.com/akiomik/cats-nio-file

Reply:
I think that's a nice change but I'd rather do it in a separate PR, this is is already taking forever 😓

Reply:
Yeah I got the easy part migrating files 😂